### PR TITLE
CI-104 capital invest opportunity pages

### PR DESCRIPTION
**Pull request by jessgilbert**
screencapture-international-trade-great-8012-international-content-capital-invest-2019-05-21-15_42_26 (1)
screencapture-international-trade-great-8012-international-content-midlands-engine-2019-05-21-15_43_51 (1)
screencapture-international-trade-great-8012-international-content-midlands-engine-housing-2019-05-21-15_44_19 (1)
screencapture-international-trade-great-8012-international-content-opportunities-ashton-green-2019-05-21-15_44_34 (1)

### CODE REVIEW

File: core/templates/core/capital_invest_landing_page.html [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+{% if page.contact_section_title and page.contact_section_text %}
+<section class="capital-invest-landing-page-section">
+	<div class="container">
+		<div class="card-grid">
+			<div class="column-half">
+				<h2 class="heading-large">{{ page.contact_section_title }} </h2>
+				<p class="font-small">{{ page.contact_section_text }} </p>
+				{% if page.contact_section_cta_text %}
+                    <div>
+                        <a href="/contact" class="button button-arrow-small">{{ page.contact_section_cta_text }} </a>
```
Comment (richtier) [inline core/templates/core/capital_invest_landing_page.html]: can you use `{% url ...` please, or `{{ services_urls.contact_us }}` context variable (whichever is relevant

File: core/templates/core/capital_invest_opportunity_page.html [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		</div>
+	</div>
+</section>
+
+<section class="capital-invest-opportunity-page-section capital-invest-region-icons-section">
+	<div class="container">
+		<div class="grid-row padding-bottom-90 ">
+		{% if page.location %}
+			<div class="column-third">
+				<img src="https://static.thenounproject.com/png/28190-200.png" class="icon">
```
Comment (richtier) [inline core/templates/core/capital_invest_opportunity_page.html]: please download those files and add them to our version control. generally the fewer external dependencies the better :)

File: core/views.py [hunk 3]
```diff
@@ -52,6 +53,28 @@ def get_context_data(self, **kwargs):
 
         context = super().get_context_data(page=self.page, **kwargs)
 
+        if self.page['page_type'] == 'CapitalInvestRegionPage' and \
```
Comment (richtier) [inline core/views.py]: as there are quite a few flags you could do something like:
```
flag_map = {
    'CapitalInvestRegionPage': 'CAPITAL_INVEST_REGION_PAGE_ON',
    'CapitalInvestRegionalSectorPage': 'CAPITAL_INVEST_REGIONAL_SECTOR_PAGE_ON',
}
...
flag_name = flag_map.get(self.page['page_type'])

if flag_name and not settings.FEATURE_FLAGS[flag_name]:
    raise Http404
```

File: core/sass/main.scss [hunk 4]
```diff
@@ -22,6 +22,10 @@
 // to prevent style pollution
 
 @import 'pages/landing-page';
+@import 'pages/capital-invest-landing-page';
+@import 'pages/_capital_invest_region_page';
```
Comment (MichaelWalker) [inline core/sass/main.scss]: This file is the only one that starts with an underscore - is that deliberate?

File: core/sass/pages/_capital-invest-landing-page.scss [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        .energy_pdf_button {
+          width: 100%;
+        }
+      }
+    }
+  }
+
+  @media (max-width: 1180px) and (min-width: 641px) {
+    .column-third-xl {
```
Comment (MichaelWalker) [inline core/sass/pages/_capital-invest-landing-page.scss]: This looks slightly surprising. I'd expect the `column-third-<size>` pattern to handle break points and things for us. Is there a reason that it doesn't work for us here?

File: core/sass/pages/_capital-invest-landing-page.scss [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+      width: 50%;
+    }
+  }
+
+  @media (max-width: 450px) {
+    .hero {
```
Comment (MichaelWalker) [inline core/sass/pages/_capital-invest-landing-page.scss]: I think we now have this hero banner in directory_components (so hopefully don't need to do anything custom here)
Could we use that version instead?

File: core/sass/pages/_capital-invest-opportunity-page.scss [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    padding: $spacing-1 0;
+  }
+
+  .capital-invest-opportunity-page-section:nth-child(odd) {
+    @include landingPageSection;
+  }
+
+  .capital-invest-opportunity-page-section:nth-child(even) {
+    @include landingPageSection;
+    background: $stone-40;
```
Comment (MichaelWalker) [inline core/sass/pages/_capital-invest-opportunity-page.scss]: I like this pattern, and we use it quite a few times. Could we make a SASS mixin for it?

File: core/sass/pages/_capital_invest_region_page.scss [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    @include landingPageSection;
+  }
+
+  .capital-invest-region-page-section:nth-child(even) {
+    @include landingPageSection;
+    background: $stone-40;
+  }
+
+  .icon {
+      height: 64px;
```
Comment (MichaelWalker) [inline core/sass/pages/_capital_invest_region_page.scss]: We also use this icon height in quite a few places. Is there already an `$icon-height` variable somewhere that we can use? (If not perhaps we should make one)

(If I'm wrong and this isn't a standard size for our icons then just ignore me! 😊)

File: core/sass/pages/_landing-page.scss [hunk 9]
```diff
@@ -60,108 +60,154 @@
     }
   }
 
-  .tariffs-section {
-    .country-selector-dialog {
-      border: 0;
+  .landing-page-section:nth-child(odd) {
```
Comment (MichaelWalker) [inline core/sass/pages/_landing-page.scss]: I wasn't expecting the international landing page to change much in this commit. Am I missing something?

File: core/templates/core/capital_invest_landing_page.html [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+						<img class="width-full" src="{{page.energy_sector_image.url}}">
+					</figure>
+				{%  endif %}
+			</div>
+		</div>
+  </div>
+</section>
+{% endif %}
+
+{% if page.section_title and page.section_content %}
```
Comment (MichaelWalker) [inline core/templates/core/capital_invest_landing_page.html]: `section_title` and `section_content` are quite generic names. Could you make them a little more descriptive?

File: core/templates/core/capital_invest_opportunity_page.html [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+{% load card hero add_export_elements_classes case_study from directory_components_tags %}
+
+{% block css_layout_class %}capital-invest-opportunity-page{% endblock %}
+
+{% block content %}
+
+{% include 'directory_components/hero.html' with hero_text=page.hero_title background_image_url=page.hero_image.url %}
+
+<section id="breadcrumb-section">
+  <div class="container">
```
Comment (MichaelWalker) [inline core/templates/core/capital_invest_opportunity_page.html]: There's now a breadcrumbs thing in directory_components.
Would that work for us here?

File: core/templates/core/capital_invest_opportunity_page.html [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+	<div class="container">
+	    {% if page.project_background_title and page.project_background_intro %}
+		<div class="grid-row">
+			<h2 class="heading-large">{{ page.project_background_title }} </h2>
+			<div class="font-medium padding-bottom-30 column-half-l">
+				{{ page.project_background_intro}}
+			</div>
+		</div>
+		{% endif %}
+		<div class="grid-row">
```
Comment (MichaelWalker) [inline core/templates/core/capital_invest_opportunity_page.html]: I think the `<div/>`s here don't close correctly for all variations of the `if`s??

This open tag is always written, but it looks as though its only closed if in the `if page.project_promoter_title` block is used.

File: core/templates/core/capital_invest_sector_page.html [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+        <li>
+          <a href="{% url 'index' %}">Great.gov.uk International</a>
+        </li>
+        <li>
+          <a href="{% url 'cms-page-from-path' 'capital-invest' %}">Capital Invest</a>
+        </li>
+        {% if page.parent %}
+			<li>
+			  <a href="{% url 'cms-page-from-path' page.parent.meta.slug %}">{{ page.parent.hero_title }}</a>
+			</li>
```
Comment (MichaelWalker) [inline core/templates/core/capital_invest_sector_page.html]: Does this optional extra parent prevent us from using the directory_components breadcrumb?
That would be sad if so! 😢

**Pull request merged**